Blazor.Test/Shared/MainLayout.razor

```diff
@@ -0,0 +1,130 @@
+@inject NavigationManager Navigation
+
+    <CDTSLayout RefTop = "@refTopObj" Top = "@topObj" PreFooter = "@preFooterObj" Footer = "@footerObj" RefFooter = "@refFooterObj" LeftMenu = "@leftMenuObj">
+        <NavLink href="" Match="NavLinkMatch.All">
+            <button class="btn">Home</button>
+        </NavLink>
+        <NavLink href="externallink">
+            <button class="btn">External Link</button>
+        </NavLink>
+        <NavLink href="wetform">
+            <button class="btn">Wet Form</button>
+        </NavLink>
+            
+        <button class="btn" @onclick="AddSearch">Add Search</button>
+        <button class="btn" @onclick="AddFooter">Add Footer</button>
+        <button class="btn" @onclick="AddLeftMenu">Add left Menu</button>  
+        <button class="btn" @onclick="AddShareBtn">Add Share</button>  
+        @Body
+    </CDTSLayout>
+
+@code{
+
+    //Create your objects
+    GoC.WebTemplate.Blazor.Model.RefTop refTopObj = new GoC.WebTemplate.Blazor.Model.RefTop()
+    {
+        //For an application template, set IsApplication to true:
+        //IsApplication = true
+    };
+
+    //For an application template, change the following and create an appTop object instead: 
+    GoC.WebTemplate.Blazor.Model.Top topObj = new GoC.WebTemplate.Blazor.Model.Top()
+    {
+        Breadcrumbs = new List<Breadcrumb>() { new Breadcrumb() { Href = "", Title = "CDTS", Acronym = "CDTS" } },
+        SiteMenu = true //For an application template, comment this line
+        //topObj.AppName = new List<Link>() { new Link() { Text = "App name" } }; //For an application template, set your application name
+    };
+
+    GoC.WebTemplate.Blazor.Model.PreFooter preFooterObj = new GoC.WebTemplate.Blazor.Model.PreFooter()
+    {
+        DateModified = "Feb 27th, 2024",
+        ShowShare = new ShareList() { Show = false }
+    };
+
+    //For an application template, change the following and create an appFooter object instead:
+    GoC.WebTemplate.Blazor.Model.Footer footerObj = new GoC.WebTemplate.Blazor.Model.Footer();
+
+    GoC.WebTemplate.Blazor.Model.RefFooter refFooterObj = new GoC.WebTemplate.Blazor.Model.RefFooter()
+    {
+        ExitSecureSite = new ExitSecureSite()
+        {
+            ExitScript = true,
+            DisplayModal = true
+        }
+    };
+
+    List<GoC.WebTemplate.Components.Entities.MenuSection> leftMenuObj;
+
+    protected override Task OnInitializedAsync()
+    {
+        JS.InvokeVoidAsync("SetCurrentPage", Navigation.ToBaseRelativePath(Navigation.Uri));
+        
+        List<LanguageLink> links = new List<LanguageLink>()
+        {
+            new LanguageLink(){Href=Navigation.BaseUri + "ChangeLang"}
+        };        
+
+        topObj.LngLinks = links;
+
+        //Detect when the user has clicked on another page
+        Navigation.LocationChanged += (o, e) => {
+            JS.InvokeVoidAsync("SetCurrentPage", Navigation.ToBaseRelativePath(Navigation.Uri));
+        };
```

Comment:
As discussed, we can have a `HandleLanguageLink` function here that:
- Does the 2 JS.Invoke calls
- Does NOT do the language links setup in topObj.  Instead, when we call Top we can optionally override LngLinks following these rules:
  - If LingLinks is NOT null, we don't touch. It means they set a custom URL or an empty array meaning no language links
  - If LngLinks is null, we add our default one

Reply:
Done (created a HandleLanguageLink component instead that we call)